crates/iceberg-catalog/src/request_metadata.rs

```diff
@@ -11,13 +11,19 @@ use uuid::Uuid;
 use crate::{service::authn::Actor, ProjectIdent, DEFAULT_PROJECT_ID};
 
 pub const PROJECT_ID_HEADER: &str = "x-project-ident";
+pub const X_REQUEST_ID_HEADER: &str = "x-request-id";
+
+const X_FORWARDED_FOR_HEADER: &str = "x-forwarded-for";
+const X_FORWARDED_PROTO_HEADER: &str = "x-forwarded-proto";
+const X_FORWARDED_PORT_HEADER: &str = "x-forwarded-port";
 
 /// A struct to hold metadata about a request.
 #[derive(Debug, Clone)]
 pub struct RequestMetadata {
     request_id: Uuid,
     project_id: Option<ProjectIdent>,
     authentication: Option<Authentication>,
+    host: Option<String>,
```

Comment:
I think we should rename this everywhere to `base_url` or so - a host comes without protocol, just like the HOST header doesn't have the protocol prepended.
This would then also use the same naming as the config option that can be used to override the host.